test/webaudio/audio_worklet_tone_generator.c

```diff
@@ -38,12 +40,12 @@ bool ProcessAudio(int numInputs, const AudioSampleFrame *inputs, int numOutputs,
 
   // Produce a sine wave tone of desired frequency to all output channels.
   for(int o = 0; o < numOutputs; ++o)
-    for(int i = 0; i < 128; ++i)
+    for(int i = 0; i < outputs[o].quantumSize; ++i)
     {
       float s = emscripten_math_sin(phase);
       phase += phaseIncrement;
       for(int ch = 0; ch < outputs[o].numberOfChannels; ++ch)
-        outputs[o].data[ch*128 + i] = s * currentVolume;
+        outputs[o].data[ch*outputs[o].quantumSize + i] = s * currentVolume;
```

Comment:
Given that the quantum size is a constant, I wonder if it would be better to have all these examples use that global `quantumSize` integer, e.g.

```c
int quantumSize;

bool ProcessAudio(int numInputs, ...) {
  for(int o = 0; o < numOutputs; ++o)
    for(int i = 0; i < quantumSize; ++i)
      ...
}

int main() {
  quantumSize = emscripten_audio_context_quantum_size(context);
}
```
This would communicate the invariant that the quantumSize cannot change between inputs and outputs in a single call to ProcessAudio(). That might help someone learning about AudioWorklets.

Alternatively, I wonder if the field `AudioSampleFrame::quantumSize` might be simpler to rename to `AudioSampleFrame::length` to signal that it just means the length of the data array. Then users could have

```c
for(int i = 0; i < outputs[o].length; ++i) {
  outputs[o].data[i] = ...;
```

with very idiomatic reading C code. The "quantumSize" name in this context reads fancier than it necessarily needs to be (and doesn't match the Web Audio spec which calls it "renderQuantumSize").

Works either way, leaving it up to you to decide.

Reply:
I don't like setting it as a global, and I agree, I also don't like the name `quantumSize` either (or `renderQuantumSize` or the use of quantum in general for this, or the way audio frame is used, etc.).

If I'm going to spend more time with this I'd rather that be spent marking the existing API as deprecated and creating a new callback signature.

Reply:
Gotcha, that makes sense. 👍